sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -352,8 +356,16 @@ private[sql] class RocksDBStateStoreProvider
     var defaultColFamilyId: Option[Short] = None
     if (useColumnFamilies) {
       // put default column family only if useColumnFamilies are enabled
-      colFamilyNameToIdMap.putIfAbsent(StateStore.DEFAULT_COL_FAMILY_NAME, colFamilyId.shortValue())
+      logError(s"### colFamilyIds: $columnFamilyIds")
+      colFamilyNameToIdMap.putAll(columnFamilyIds.asJava)
       defaultColFamilyId = Option(colFamilyId.shortValue())
+      columnFamilyIds.filter(_ != null).map { case (colFamilyName, colFamilyId) =>
+        logError(s"### Putting colFamilyName: $colFamilyName with id $colFamilyId")
+        keyValueEncoderMap.putIfAbsent(colFamilyName,
+          (RocksDBStateEncoder.getKeyEncoder(keyStateEncoderSpec, useColumnFamilies,
+            Some(colFamilyId)), RocksDBStateEncoder.getValueEncoder(valueSchema,
+            useMultipleValuesPerKey)))
```

Comment:
We'll need to increment `colFamilyId` here; if after restart we call `createColFamilyIfExists`, we will need to avoid Id collisions

Reply:
On second thought, there probably won't be any situation where we will call `createColFamilyIfExists` after restart.